server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1038,13 +1039,31 @@ synchronized void completeClose(boolean saveState, boolean completeClose) throws
       activeScan.interrupt();
     }
 
+    // create a copy so that it can be whittled down as client sessions are disabled
```

Comment:
The call to `ScanDataSource.interrupt()` on line 1039 above sets the interrupt flag on the iterators to true. The issue we are addressing here is zombie scans, where the interrupt flag is being set, but not checked in some user supplied iterator (#4757). I wonder if we want to capture the time that the interrupt flag was set to true, and wait some period of time before forcing the tablet to close giving scans that are behaving correctly a chance to succeed.

Reply:
> I wonder if we want to capture the time that the interrupt flag was set to true, and wait some period of time before forcing the tablet to close

I was initially going to wait but decided against it for the following reasons.

 1. Once the interrupt flag is set its a matter of luck if the scan threads gets additional data or not.  So waiting may not result in clients getting additional data.
 2. Since the tablet no longer allows new scans to start, we do not know how many of those are currently being held up.
 3. Attempting to close the clients scan session will naturally introduce some delay because of the way reservations work on scan sessions.  Will wait for any client currently holding a reservation to relinquish it, which clients do every second by default but this is configurable.  After a client does relinquish the reservation on the session it can never be reserved again with the new disable code added in this PR.